src/models/appmodel.ts

```diff
@@ -11,17 +12,39 @@ export class AppModel implements IAppModel {
 
     @observable private _selectedFlavor = "";
     get selectedFlavor(): string { return this._selectedFlavor; }
-    set selectedFlavor(item: string) { this._selectedFlavor = item; }
+    set selectedFlavor(item: string) { 
+        this._selectedFlavor = item; 
+        if(!this._isConstructing) this.saveState();
+    }
 
+    private _isConstructing = true;q
     @observable private _textInput = "";
     get flavorInput(): string { return this._textInput; }
     set flavorInput(value: string) { this._textInput = value; }
 
     get flavorTextIsValid() { return Boolean(this.flavorInput); }
 
-    constructor()
+    private _dataModel: IDataModel;
+
+    constructor(dataModel: IDataModel)
     {
         this.selectedFlavor = this.flavors[0];
+        this._dataModel = dataModel;
+        const data = dataModel.load();
+        if(data != null && data != "")
```

Comment:
FYI
it's always better to use a type checking as well regarding luck of strict types in JS | TS. It's just a very nice habit
https://www.w3schools.com/js/js_comparisons.asp

also in this case, if it isn't important for us if the data NULL etc, we could use just `if (data) {...}`   or ` if (Boolean(data)) {...}` just if we want to increase sanity a bit

Reply:
in this, if condition I personally prefer just to use  ` if (data) {....}`
because it's simpler than adding a strict boolean transformation. The ts will do it automatically.